feature/gribi/otg_tests/gribigo_compliance_test/gribigo_compliance_test.go

```diff
@@ -225,3 +290,67 @@ func configureATE(t *testing.T, ate *ondatra.ATEDevice) gosnappi.Config {
 
 	return top
 }
+
+// AddNHGReferencingToUknownNH tests a case where a nexthop group references a nexthop IP that does not Resolve.
+func AddNHGReferencingToUnresolvedNH(tcArgs *testArgs, wantACK fluent.ProgrammingResult, t testing.TB) {
+	t.Log("Flush existing gRIBI routes before test.")
+	if err := gribi.FlushAll(tcArgs.client); err != nil {
+		t.Fatal(err)
+	}
+	tcArgs.client.Modify().AddEntry(t,
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(tcArgs.dut)).
+			WithIndex(2000).WithIPAddress("1.0.0.1"),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(tcArgs.dut)).
+			WithID(2000).AddNextHop(2000, 1),
+		fluent.IPv4Entry().WithNetworkInstance(deviations.DefaultNetworkInstance(tcArgs.dut)).
+			WithPrefix("203.0.113.101/32").WithNextHopGroup(2000))
+
+	if err := awaitTimeout(tcArgs.ctx, t, tcArgs.client, 2*time.Minute); err != nil {
+		t.Logf("Could not program entries via client, got err, check error codes: %v", err)
+	}
+	chk.HasResult(t, tcArgs.client.Results(t),
+		fluent.OperationResult().
+			WithNextHopGroupOperation(2000).
+			WithOperationType(constants.Add).
+			WithProgrammingResult(wantACK).
+			AsResult(),
+		chk.IgnoreOperationID())
+}
+
+// AddNHGReferencingToDownPort tests a case where a nexthop group references to nexthop port that is down.
+func AddNHGReferencingToDownPort(tcArgs *testArgs, wantACK fluent.ProgrammingResult, t testing.TB) {
+	t.Log("Setting port2 to down...")
+	p := tcArgs.dut.Port(t, "port2")
+	t.Log("Flush existing gRIBI routes before test.")
+	if err := gribi.FlushAll(tcArgs.client); err != nil {
+		t.Fatal(err)
+	}
+	setDUTInterfaceWithState(t, tcArgs.dut, p, false)
+	tcArgs.client.Modify().AddEntry(t,
+		fluent.NextHopEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(tcArgs.dut)).
+			WithIndex(2000).WithInterfaceRef(p.Name()).WithIPAddress(atePort2.IPv4),
+		fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(tcArgs.dut)).
+			WithID(2000).AddNextHop(2000, 1),
+		fluent.IPv4Entry().WithNetworkInstance(deviations.DefaultNetworkInstance(tcArgs.dut)).
+			WithPrefix("203.0.113.100/32").WithNextHopGroup(2000),
+	)
+	if err := awaitTimeout(tcArgs.ctx, t, tcArgs.client, 2*time.Minute); err != nil {
+		t.Logf("Could not program entries via client, got err, check error codes: %v", err)
```

Comment:
Similarly here, should this be an `Errorf`?

Reply:
this is fine we are printing the log only with and checking the below where actually it is failing/passing.

chk.HasResult(t, tcArgs.client.Results(t),